lib_nbgl/src/nbgl_use_case.c

```diff
@@ -1526,5 +1547,197 @@ void nbgl_useCaseSpinner(const char *text)
     pageContext = nbgl_pageDrawSpinner(NULL, (const char *) text);
     nbgl_refreshSpecial(FULL_COLOR_PARTIAL_REFRESH);
 }
+
+#ifdef NBGL_KEYPAD
+/**
+ * @brief internal function to refresh the keypad and automatically show / hide:
+ *        - backspace key if no digits are present
+ *        - validation key if the min digit is reached
+ *        - keypad if the max number of digit is reached
+ *
+ * @param add indicate if a digit has been added or removed
+ */
+static void nbgl_refreshKeyPad(bool add)
+{
+    bool                enableValidate, enableBackspace, enableDigits;
+    bool                redrawKeypad = false;
+    nbgl_refresh_mode_t mode         = BLACK_AND_WHITE_FAST_REFRESH;
+
+    enableDigits    = (keypadContext.pinLen < keypadContext.pinMaxDigits);
+    enableValidate  = (keypadContext.pinLen >= keypadContext.pinMinDigits);
+    enableBackspace = (keypadContext.pinLen > 0);
+    if (add) {
+        if ((keypadContext.pinLen == keypadContext.pinMinDigits)
+            ||  // activate "validate" button on keypad
+            (keypadContext.pinLen == keypadContext.pinMaxDigits)
+            ||                              // deactivate "digits" on keypad
+            (keypadContext.pinLen == 1)) {  // activate "backspace"
+            redrawKeypad = true;
+        }
+    }
+    else {                                  // remove
+        if ((keypadContext.pinLen == 0) ||  // deactivate "backspace" button on keypad
+            (keypadContext.pinLen == (keypadContext.pinMinDigits - 1))
+            ||  // deactivate "validate" button on keypad
+            (keypadContext.pinLen
+             == (keypadContext.pinMaxDigits - 1))) {  // reactivate "digits" on keypad
+            redrawKeypad = true;
+        }
+    }
+    nbgl_layoutUpdateHiddenDigits(
+        keypadContext.layoutCtx, keypadContext.hiddenDigitsIndex, keypadContext.pinLen);
+    if (redrawKeypad) {
+        nbgl_layoutUpdateKeypad(keypadContext.layoutCtx,
+                                keypadContext.keypadIndex,
+                                enableValidate,
+                                enableBackspace,
+                                enableDigits);
+    }
+
+    if ((!add) && (keypadContext.pinLen == 0)) {
+        // Full refresh to fully clean the bullets when reaching 0 digits
+        mode = FULL_COLOR_REFRESH;
+    }
+    nbgl_refreshSpecialWithPostRefresh(mode, POST_REFRESH_FORCE_POWER_ON);
+}
+
+/**
+ * @brief keypad callback when a key is touched
+ *        - backspace key
+ *
+ * @param touchedKey the key touched
+ */
+static void nbgl_keypadCallback(char touchedKey)
```

Comment:
to be complient with all NBGL files, put static functions before all global functions

Reply:
Done